uefi-raw/src/protocol/ata.rs

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use super::device_path::DevicePathProtocol;
+use crate::Status;
+use core::ffi::c_void;
+use uguid::{guid, Guid};
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct AtaPassThruMode {
+    pub attributes: u32,
+    pub io_align: u32,
+}
+
+newtype_enum! {
+    /// Corresponds to the `EFI_ATA_PASS_THRU_PROTOCOL_*` defines.
+    #[derive(Default)]
+    pub enum AtaPassThruCommandProtocol: u8 => {
+        ATA_HARDWARE_RESET = 0x00,
+        ATA_SOFTWARE_RESET = 0x01,
+        ATA_NON_DATA = 0x02,
+        PIO_DATA_IN = 0x04,
+        PIO_DATA_OUT = 0x05,
+        DMA = 0x06,
+        DMA_QUEUED = 0x07,
+        DEVICE_DIAGNOSTIC = 0x08,
+        DEVICE_RESET = 0x09,
+        UDMA_DATA_IN = 0x0A,
+        UDMA_DATA_OUT = 0x0B,
+        FPDMA = 0x0C,
+        RETURN_RESPONSE = 0xFF,
+    }
+}
+
+newtype_enum! {
+    /// Corresponds to the `EFI_ATA_PASS_THRU_LENGTH_*` defines.
+    #[derive(Default)]
+    pub enum AtaPassThruLength: u8 => {
+        BYTES = 0x80,
+        MASK = 0x70,
+        NO_DATA_TRANSFER = 0x00,
+        FEATURES = 0x10,
+        SECTOR_COUNT = 0x20,
+        TPSIU = 0x30,
+        COUNT = 0x0F,
+    }
+}
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct AtaPassThruStatusBlock {
```

Comment:
In the spec this is named `EFI_ATA_STATUS_BLOCK`, so let's call it `AtaStatusBlock` here.